src/types/UiaTextRangeBase.cpp

```diff
@@ -990,1534 +902,359 @@ const COORD UiaTextRangeBase::_getScreenFontSize() const
-                         buffer.GetWordStart(target, wordDelimiters, true);
-
-            currentColumn = target.X;
-        }
-        *pAmountMoved += static_cast<int>(moveState.Increment);
-
-        FAIL_FAST_IF(!(currentColumn >= _getFirstColumnIndex()));
-        FAIL_FAST_IF(!(currentColumn <= _getLastColumnIndex(pData)));
-        FAIL_FAST_IF(!(currentScreenInfoRow >= _getFirstScreenInfoRowIndex()));
-        FAIL_FAST_IF(!(currentScreenInfoRow <= _getLastScreenInfoRowIndex(pData)));
-    }
-
-    Endpoint start = _screenInfoRowToEndpoint(pData, currentScreenInfoRow) + currentColumn;
-
-    auto target = _endpointToCoord(pData, start);
-    target = buffer.GetWordEnd(target, wordDelimiters, true);
-    Endpoint end = _coordToEndpoint(pData, target);
-
-    return std::make_pair<Endpoint, Endpoint>(std::move(start), std::move(end));
-}
-
-// Routine Description:
-// - calculates new Endpoints if they were to be moved moveCount times
-// by line.
-// Arguments:
-// - pData - the UiaData for the terminal we are operating on
-// - moveCount - the number of times to move
-// - moveState - values indicating the state of the console for the
-// move operation
-// - pAmountMoved - the number of times that the return values are "moved"
-// Return Value:
-// - a pair of endpoints of the form <start, end>
-std::pair<Endpoint, Endpoint> UiaTextRangeBase::_moveByLine(gsl::not_null<IUiaData*> pData,
-                                                            const int moveCount,
-                                                            const MoveState moveState,
-                                                            _Out_ gsl::not_null<int*> const pAmountMoved)
-{
-    *pAmountMoved = 0;
-    Endpoint start = _screenInfoRowToEndpoint(pData, moveState.StartScreenInfoRow) + moveState.StartColumn;
-    Endpoint end = _screenInfoRowToEndpoint(pData, moveState.EndScreenInfoRow) + moveState.EndColumn;
-    ScreenInfoRow currentScreenInfoRow = moveState.StartScreenInfoRow;
-    // we don't want to move the range if we're already in the
-    // limiting row and trying to move off the end of the screen buffer
-    const bool illegalMovement = (currentScreenInfoRow == moveState.LimitingRow &&
-                                  ((moveCount < 0 && moveState.Increment == MovementIncrement::Backward) ||
-                                   (moveCount > 0 && moveState.Increment == MovementIncrement::Forward)));
-
-    if (moveCount != 0 && !illegalMovement)
-    {
-        // move the range
-        for (int i = 0; i < abs(moveCount); ++i)
-        {
-            if (currentScreenInfoRow == moveState.LimitingRow)
+            if (nextPos == bufferEnd)
             {
-                break;
+                fSuccess = false;
+            }
+            else if (buffer.MoveToNextWord(nextPos, _wordDelimiters, lastCharPos))
+            {
+                resultPos = nextPos;
+                *pAmountMoved += 1;

```

Comment:
This is weird versus `++`, isn't it? I've seen it a bunch of times now.

Reply:
![image](https://user-images.githubusercontent.com/11050425/73424893-ed557980-42e4-11ea-9b9c-9eab592877f2.png)

I'm literally cursing at my office right now

Reply:
Thanks @rioshin!